nima/http2/webserver/src/main/java/io/helidon/nima/http2/webserver/Http2Connection.java

```diff
@@ -624,6 +627,13 @@ private void doHeaders(Semaphore requestSemaphore) {
                                           requestHuffman,
                                           new Http2FrameData(frameHeader, inProgressFrame()));
         }
+        ctx.remotePeer().tlsCertificates()
```

Comment:
The X_HELIDON_CN header should not be created for each request on this connection. The remote peer stays the same for the duration of the connection, so we should create the header just once, and then reuse it. 
Also updating the headers later on is not a good choice (also because you are doing an instanceof), so I think this should be part of creation of the Http2Headers (maybe add `Headers` instance as a parameter and copy all headers from that instance to the new instance being created)